x-pack/plugins/fleet/public/applications/integrations/hooks/use_local_search.tsx

```diff
@@ -16,13 +16,11 @@ export const fieldsToSearch = ['name', 'title'];
 export function useLocalSearch(packageList: IntegrationCardItem[]) {
   const localSearchRef = useRef<LocalSearch>(new LocalSearch(searchIdField));
 
-  useEffect(() => {
```

Comment:
I am not sure why this useEffect was neccesary? is it best practice to only change ref.current in a useEffect? I can't find anything which says so.

The bug was a race condition, the results are rendered before the useEffect is called, meaning the ref contains the stale search resaults. Removing the useEffect means the results are instantly updated removing the race.